src/com/sun/ts/tests/jaspic/tssv/config/TSAuthConfigProvider.java

```diff
@@ -214,8 +214,16 @@ public ServerAuthConfig getServerAuthConfig(String layer, String appContext,
         logger.log(Level.INFO, msg);
       }
 
-      ServerAuthConfig serverAuthConfig = new TSServerAuthConfig(layer,
-          appContext, handler, properties, logger);
+      ServerAuthConfig serverAuthConfig = null;
+
+      if(JASPICData.LAYER_SOAP.equals(layer)) {
+          serverAuthConfig = new SOAPTSServerAuthConfig(layer,
```

Comment:
Nice fix, it might of been nice to have a comment that mentions which one is for Web Profile and which is for Full Platform but not required.